scipy/_lib/_array_api.py

```diff
@@ -171,23 +171,35 @@ def _asarray(
 
     `subok` is included to allow this function to preserve the behaviour of
     `np.asanyarray` for NumPy based inputs.
+
     """
+    xp_in = array_namespace(array)
     if xp is None:
-        xp = array_namespace(array)
-    if is_numpy(xp):
-        # Use NumPy API to support order
-        if copy is True:
-            array = np.array(array, order=order, dtype=dtype, subok=subok)
-        elif subok:
-            array = np.asanyarray(array, order=order, dtype=dtype)
-        else:
-            array = np.asarray(array, order=order, dtype=dtype)
+        xp = xp_in
+
+    if is_numpy(xp_in):
+        # If object is array-like (but not array), make it an ndarray; otherwise, no-op.
+        array = np.asanyarray(array)
+
+    if is_numpy(xp_in) and not array.__class__.__name__.endswith('.ndarray') and subok:
```

Comment:
Replace with preferred method of determining whether object is strict subclass of ndarray.